### PR TITLE
board: Make the storage root directory optional

### PR DESCRIPTION
**Pull request by muramura**
I know several PRs who do APM to AP.
The directory name of the storage is APM.
I make the storage root directory name AP.

### CODE REVIEW

**Comment by IamPete1** [comment]
I do think this is a good idea, but to just change it will mean everyone's scripts and terrain will stop being used. Maybe it should try and re-name?

File: libraries/AP_HAL_ChibiOS/hwdef/CUAV-Nora/hwdef.dat [hunk 1]
```diff
@@ -294,8 +294,8 @@ PE10 SAFETY_IN INPUT PULLDOWN
 # enable FAT filesystem support (needs a microSD defined via SDMMC)
 define HAL_OS_FATFS_IO 1
 
-define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
-define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+define HAL_BOARD_LOG_DIRECTORY HAL_BOARD_STORAGE_DIRECTORY "/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY HAL_BOARD_STORAGE_DIRECTORY "/TERRAIN"
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/CUAV-Nora/hwdef.dat:298]: This identical patch exists in *how* many hwdef.dat files?!

Something is wrong here.  Perhaps this should be added into the hwdef.py script instead - or just removed after verifying that this is the default value anyway!

**Review by peterbarker — CHANGES_REQUESTED** [review]
Please split the change-of-directory into a separate PR - that's going to break things and will require a lot of thought!

Most of the patches in here are good cleanups - especially if we can simply remove all of those lines from the hwdef

File: libraries/AP_HAL_ChibiOS/hwdef/CUAV-Nora/hwdef.dat (hunk 1 above)
Comment (muramura) [reply libraries/AP_HAL_ChibiOS/hwdef/CUAV-Nora/hwdef.dat:298]: I installed it on my CONTROL ZERO CLASSIC.
I am able to confirm that this change is effective.

Screenshot from 2021-09-19 11-42-20

Screenshot from 2021-09-19 11-42-36

**Comment by muramura** [comment]
@peterbarker san.
I'll split it into two PRs.
I will make sure that this HAL_BOARD_STORAGE_DIRECTORY definition determines the root directory for all FCs.
I will make it so that the root directory can be easily changed by changing this definition.

**Comment by muramura** [comment]
This PR has been split into #18711 and #18712.
Therefore, this PR is closed.

**Pull request closed without merging**